### ISSUE TITLE
get_solarpostion in modelchain.prepare_inputs does not use weather data.

### ISSUE THREAD

**Issue by kenkuwae** [pvlib/pvlib-python#523]
**Problem**
``get_solarpostion`` method in ``modelchain.ModelChain.prepare_inputs`` does not use weather data.

**To Reproduce**
change line 745 - 746
``` modelchain.py
755:       self.solar_position = self.location.get_solarposition(
756:            self.times, method=self.solar_position_method)
```

to 

``` modelchain.py
755:       self.solar_position = self.location.get_solarposition(
756:            self.times, method=self.solar_position_method,
757:            pressure=self.weather.pressure,
758:            temperature=self.weather.temp_air)
```

**Versions:**
 - ``pvlib.__version__``:  0.5.2
 - ``pandas.__version__``: 0.23.3
 - python: 3.5.2

**Comment by wholmgren** [comment]
Good idea to pass this data on to get_solarposition if available. What about the case where one or more is not available? Maybe use pvlib.tools._build_kwargs? 

Pull request welcome. Might not happen otherwise.

**Comment by kenkuwae** [comment]
thank you for response.
ok, I consider how to do that.
If I found any idea, I share this.

**Comment by cwhanse** [comment]
@kenkuwae thanks for opening the issue.  I am re-opening so that we can keep this enhancement on the list.

**Comment by Ahanmr** [comment]
@kenkuwae  @cwhanse  I have taken up this issue, and fixed the changes and applied for a PR.

**Comment by wholmgren** [comment]
closed by #936, but possibly a bug.
